crates/solidity/testing/snapshots/cst_output/SourceUnit/pratt_precedence_recovery/generated/0.4.11-failure.yml

```diff
@@ -0,0 +1,57 @@
+  5  │ }                                                                                │ 40..41
+
+Errors: # 1 total
+  - >
+    Error: Expected CloseParen or Comma.
+       ╭─[crates/solidity/testing/snapshots/cst_output/SourceUnit/pratt_precedence_recovery/input.sol:4:2]
+       │
+     4 │     }
+       │     │ 
+       │     ╰─ Error occurred here.
+    ───╯
+
+Tree:
+  - (SourceUnit) ► (members꞉ SourceUnitMembers): # "contract A {\n\tfunction a() {\n\t\t2 + (\n\t" (0..38)
+      - (item꞉ SourceUnitMember) ► (variant꞉ ContractDefinition): # "contract A {\n\tfunction a() {\n\t\t2 + (\n\t" (0..38)
+          - (contract_keyword꞉ ContractKeyword): "contract" # (0..8)
+          - (leading_trivia꞉ Whitespace): " " # (8..9)
+          - (name꞉ Identifier): "A" # (9..10)
+          - (leading_trivia꞉ Whitespace): " " # (10..11)
+          - (open_brace꞉ OpenBrace): "{" # (11..12)
+          - (trailing_trivia꞉ EndOfLine): "\n" # (12..13)
+          - (members꞉ ContractMembers): # "\tfunction a() {\n\t\t2 + (" (13..36)
+              - (item꞉ ContractMember) ► (variant꞉ FunctionDefinition): # "\tfunction a() {\n\t\t2 + (" (13..36)
+                  - (leading_trivia꞉ Whitespace): "\t" # (13..14)
+                  - (function_keyword꞉ FunctionKeyword): "function" # (14..22)
+                  - (name꞉ FunctionName): # " a" (22..24)
+                      - (leading_trivia꞉ Whitespace): " " # (22..23)
+                      - (variant꞉ Identifier): "a" # (23..24)
+                  - (parameters꞉ ParametersDeclaration): # "()" (24..26)
+                      - (open_paren꞉ OpenParen): "(" # (24..25)
+                      - (parameters꞉ Parameters): [] # (25..25)
+                      - (close_paren꞉ CloseParen): ")" # (25..26)
+                  - (attributes꞉ FunctionAttributes): [] # (26..26)
+                  - (body꞉ FunctionBody) ► (variant꞉ Block): # " {\n\t\t2 + (" (26..36)
+                      - (leading_trivia꞉ Whitespace): " " # (26..27)
+                      - (open_brace꞉ OpenBrace): "{" # (27..28)
+                      - (trailing_trivia꞉ EndOfLine): "\n" # (28..29)
+                      - (statements꞉ Statements): # "\t\t2 +" (29..34)
+                          - (item꞉ Statement) ► (variant꞉ ExpressionStatement) ► (expression꞉ Expression): # "\t\t2 +" (29..34)
+                              - (variant꞉ DecimalNumberExpression): # "\t\t2" (29..32)
+                                  - (leading_trivia꞉ Whitespace): "\t\t" # (29..31)
+                                  - (literal꞉ DecimalLiteral): "2" # (31..32)
+                              - (AdditiveExpression): # " +" (32..34)
+                                  - (leading_trivia꞉ Whitespace): " " # (32..33)
+                                  - (operator꞉ Plus): "+" # (33..34)
+                      - (MISSING): "" # (34..34)
+                      - (close_brace꞉ CloseBrace): " " # (34..35)
```

Comment:
Looks like this uncovered another bug with delimiter recovery. I'll double check.

Reply:
I think the PRATT parser in this branch is producing an incomplete tree.
I created #1040 which will make the bug fail this specific test.

Reply:
Small mistake on my part, I forgot to also add `.chain(skipped.nodes)`. Thanks for noticing that!